projects/ngneat/dialog/src/lib/dialog-ref.ts

```diff
@@ -17,7 +17,7 @@ export abstract class DialogRef<
   public data: Data;
 
   public backdropClick$: Observable<MouseEvent>;
-  public afterClosed$: Observable<Result>;
+  public afterClosed$: Observable<Result | undefined>;
```

Comment:
Can you explain the changes in this file and how it's related to the types?

Reply:
iirc this is here becuase closing the dialog without any arguments (which is valid) makes the result `undefined`. I _think_ this also happens when dismissing using the close button

(and yes, if you don't type the dialog result, `Result = any` will kick in which includes `undefined`, but with a typed dialog result `afterClosed$: Observable<Result>` incorrectly suggests that the value will always be defined)

Reply:
Indeed, as @jemand771 states; the `close` method parameter is optional (see [dialog-ref.ts#L22](https://github.com/ngneat/dialog/blob/master/projects/ngneat/dialog/src/lib/dialog-ref.ts#L22)).
Following this through [InternalDialogRef.onClose](https://github.com/ngneat/dialog/blob/master/projects/ngneat/dialog/src/lib/dialog-ref.ts#L40-L44) and the [mutating of the ref](https://github.com/ngneat/dialog/blob/master/projects/ngneat/dialog/src/lib/dialog.service.ts#L177-L184), we see that the `afterClosed` hook gets called with this result (see [dialog.service.ts#L166](https://github.com/ngneat/dialog/blob/master/projects/ngneat/dialog/src/lib/dialog.service.ts#L166))

The added parenthesis are an IDE config that is too strict; I'll remove them since they are an unrelated change.

Reply:
@NetanelBasal The parenthesis later down this file seem to be added by prettier; are you okay with keeping them in?

Reply:
Yes